### PR TITLE
[folly/fizz/wangle/proxygen/fbthrift] Update to latest revision

### PR DESCRIPTION
**Pull request by PhoebeHui**
Related to https://github.com/microsoft/vcpkg/pull/13536

1. folly: remove 4 unused patches: find-double-conversion.patch fix-addbit.patch folly_c3861.patch fix-cmake-3.18.patch
2. folly: update 2 patches that match the source change in new revision.
3. folly: add fmt to 'Build-Depends'
4. proxygen: remove patch fix-tools-path.patch and use vcpkg_copy_tools instead.
4. fizz/wangle/proxygen/fbthrift need update togather.

All features test passed with x64-windows

### CODE REVIEW

File: ports/folly/portfile.cmake [hunk 1]
```diff
@@ -12,16 +12,15 @@ vcpkg_add_to_path("${PYTHON3_DIR}")
...
+    SHA512 99ba85a570a7d5b4ea3a751cca1ac447e200cff2ebfbd7265f2df9b941765e024a960b78aafe6816a2b393494d2ae1021af9573a841b99f31bfcf7d3740c9314
     HEAD_REF master
     PATCHES
         missing-include-atomic.patch
         reorder-glog-gflags.patch
         disable-non-underscore-posix-names.patch
         boost-1.70.patch
-        fix-addbit.patch
-        folly_c3861.patch
+
```
Comment (JackBoosY) [inline ports/folly/portfile.cmake:23]:
```suggestion
```

**Review by JackBoosY — CHANGES_REQUESTED** [review]

**Review by JackBoosY — APPROVED** [review]

**Comment by PhoebeHui** [comment]
proxygen:x64-osx failures looks same issue with gperf issue http://savannah.gnu.org/bugs/?53028, which fix by http://git.savannah.gnu.org/gitweb/?p=gperf.git;a=commitdiff;h=baf89d87f0d94b83e7764569da9d84dd36f82ba9.

Failures: 
```generated/proxygen/lib/http/HTTPCommonHeaders.cpp:86:20: error: ISO C++17 does not allow 'register' storage class specifier [-Wregister]
gperf_case_memcmp (register const char *s1, register const char *s2, register unsigned int n)
                   ^~~~~~~~~
generated/proxygen/lib/http/HTTPCommonHeaders.cpp:86:45: error: ISO C++17 does not allow 'register' storage class specifier [-Wregister]
gperf_case_memcmp (register const char *s1, register const char *s2, register unsigned int n)
                                            ^~~~~~~~~
generated/proxygen/lib/http/HTTPCommonHeaders.cpp:86:70: error: ISO C++17 does not allow 'register' storage class specifier [-Wregister]
gperf_case_memcmp (register const char *s1, register const char *s2, register unsigned int n)
                                                                     ^~~~~~~~~
generated/proxygen/lib/http/HTTPCommonHeaders.cpp:112:34: error: ISO C++17 does not allow 'register' storage class specifier [-Wregister]
HTTPCommonHeadersInternal::hash (register const char *str, register unsigned int len)
                                 ^~~~~~~~~
generated/proxygen/lib/http/HTTPCommonHeaders.cpp:112:60: error: ISO C++17 does not allow 'register' storage class specifier [-Wregister]
HTTPCommonHeadersInternal::hash (register const char *str, register unsigned int len)
                                                           ^~~~~~~~~
generated/proxygen/lib/http/HTTPCommonHeaders.cpp:143:3: error: ISO C++17 does not allow 'register' storage class specifier [-Wregister]
  register unsigned int hval = len;
  ^~~~~~~~~
generated/proxygen/lib/http/HTTPCommonHeaders.cpp:302:41: error: ISO C++17 does not allow 'register' storage class specifier [-Wregister]
HTTPCommonHeadersInternal::in_word_set (register const char *str, register unsigned int len)
                                        ^~~~~~~~~
generated/proxygen/lib/http/HTTPCommonHeaders.cpp:302:67: error: ISO C++17 does not allow 'register' storage class specifier [-Wregister]
HTTPCommonHeadersInternal::in_word_set (register const char *str, register unsigned int len)
                                                                  ^~~~~~~~~
generated/proxygen/lib/http/HTTPCommonHeaders.cpp:310:11: error: ISO C++17 does not allow 'register' storage class specifier [-Wregister]
          register int index = lookup[key];
          ^~~~~~~~~
generated/proxygen/lib/http/HTTPCommonHeaders.cpp:316:19: error: ISO C++17 does not allow 'register' storage class specifier [-Wregister]
                  register const char *s = wordlist[index].name;
                  ^~~~~~~~~
10 errors generated.
```

**Comment by PhoebeHui** [comment]
Disable proxygen:x64-osx temporarily for unblocking this PR, and will enable it when PR https://github.com/microsoft/vcpkg/pull/14228 merged.

**Review by JackBoosY — APPROVED** [review]

**Comment by BillyONeal** [comment]
Thanks for your contribution!

File: ports/folly/disable-non-underscore-posix-names.patch [hunk 2]
```diff
@@ -1,27 +1,34 @@
...
+ #pragma push_macro("_CRT_INTERNAL_NONSTDC_NAMES")
+-#undef _CRT_INTERNAL_NONSTDC_NAMES
 -#include <direct.h> // @manual nolint
 -#include <io.h> // @manual nolint
 -#undef __STDC__
++#define _CRT_INTERNAL_NONSTDC_NAMES 0
++#include <direct.h>
++#include <io.h>
+ #pragma pop_macro("_CRT_INTERNAL_NONSTDC_NAMES")
+-#pragma pop_macro("_CRT_DECLARE_NONSTDC_NAMES")
```
Comment (dubois) [inline ports/folly/disable-non-underscore-posix-names.patch:27]: Hi @PhoebeHui , did you mean to add this line to the .patch?  The resulting `portability/Windows.h` has a mismatched set of `pragma push/pop macro` lines.

Specifically, the resulting file contains `#pragma push_macro("_CRT_DECLARE_NONSTDC_NAMES")` but no `#pragma pop_macro("_CRT_DECLARE_NONSTDC_NAMES")`

**Pull request merged**